core/iwasm/compilation/aot_orc_extra2.cpp

```diff
@@ -0,0 +1,107 @@
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include "llvm/ExecutionEngine/Orc/CompileUtils.h"
+#include "llvm/IR/LegacyPassManager.h"
+#include "llvm/Object/ObjectFile.h"
+#include "llvm/Support/SmallVectorMemoryBuffer.h"
+#include "llvm/CodeGen/Passes.h"
+#include "llvm/CodeGen/MachineFrameInfo.h"
+#include "llvm/CodeGen/MachineFunctionPass.h"
+
+#include "bh_log.h"
+
+class MyCompiler : public llvm::orc::IRCompileLayer::IRCompiler
+{
+  public:
+    MyCompiler(llvm::orc::JITTargetMachineBuilder JTMB);
+    llvm::Expected<llvm::orc::SimpleCompiler::CompileResult> operator()(
+        llvm::Module &M) override;
+
+  private:
+    llvm::orc::JITTargetMachineBuilder JTMB;
+};
+
+MyCompiler::MyCompiler(llvm::orc::JITTargetMachineBuilder JTMB)
+  : IRCompiler(llvm::orc::irManglingOptionsFromTargetOptions(JTMB.getOptions()))
+  , JTMB(std::move(JTMB))
+{}
+
+class PrintStackSizes : public llvm::MachineFunctionPass
+{
+  public:
+    PrintStackSizes();
+    bool runOnMachineFunction(llvm::MachineFunction &MF) override;
+    static char ID;
+};
+
+PrintStackSizes::PrintStackSizes()
+  : MachineFunctionPass(ID)
+{}
+
+char PrintStackSizes::ID = 0;
+
+bool
+PrintStackSizes::runOnMachineFunction(llvm::MachineFunction &MF)
+{
+    auto name = MF.getName();
+    auto MFI = &MF.getFrameInfo();
+    size_t sz = MFI->getStackSize();
+    LOG_VERBOSE("func %.*s stack %zu", name.size(), name.data(), sz);
```

Comment:
this PR is just for a demonstration for now.
it outputs something like the following.
```
[09:06:54:689 - 70000A7C1000]: func aot_func#754_wrapper stack 0
[09:06:54:218 - 70000A7C1000]: func aot_func#758 stack 104
[09:06:54:135 - 70000A7C1000]: func aot_func#762 stack 88
[09:06:54:303 - 70000A7C1000]: func aot_func#766 stack 56
[09:06:54:190 - 70000AFC4000]: func aot_func#770 stack 88
[09:06:54:502 - 70000AFC4000]: func aot_func#770_wrapper stack 0                
[09:06:54:711 - 70000AFC4000]: func aot_func#774 stack 120     
[09:06:54:278 - 70000AFC4000]: func aot_func#778 stack 184
[09:06:54:339 - 70000AFC4000]: func aot_func#782 stack 104
[09:06:54:877 - 70000AFC4000]: func aot_func#786 stack 184
[09:06:54:192 - 70000AFC4000]: func aot_func#786_wrapper stack 0
[09:06:54:775 - 70000AFC4000]: func aot_func#790 stack 120
[09:06:54:741 - 70000AFC4000]: func aot_func#794 stack 104
[09:06:54:191 - 70000AFC4000]: func aot_func#798 stack 0
[09:06:54:074 - 700009F92000]: func aot_func#802 stack 72
```